hpx/util/thread_mapper.hpp

```diff
@@ -7,108 +8,132 @@
 #pragma once
 
 #include <hpx/config.hpp>
+#include <hpx/basic_execution/this_thread.hpp>
 #include <hpx/errors.hpp>
-#include <hpx/synchronization/spinlock.hpp>
 #include <hpx/functional/function.hpp>
+#include <hpx/synchronization/spinlock.hpp>
 
-#include <boost/bimap.hpp>
-
-#include <thread>
 #include <cstdint>
 #include <map>
 #include <string>
+#include <thread>
 #include <vector>
 
 #include <hpx/config/warnings_prefix.hpp>
 
-namespace hpx { namespace util
-{
+namespace hpx { namespace util {
+
     ///////////////////////////////////////////////////////////////////////////
     // enumerates active OS threads and maintains their metadata
-    class HPX_EXPORT thread_mapper
-    {
-    public:
-        HPX_NON_COPYABLE(thread_mapper);
 
-    public:
-        // type for callback function invoked when thread is unregistered
-        typedef util::function_nonser<bool(std::uint32_t)> callback_type;
+    class thread_mapper;
 
-        // erroneous thread index
-        static std::uint32_t const invalid_index;
-        // erroneous low-level thread ID
-        static long const invalid_tid;
-        // empty label for failed lookups
-        static const std::string invalid_label;
+    namespace detail {
 
-    private:
-        // null callback
-        static bool null_cb(std::uint32_t);
+        // type for callback function invoked when thread is unregistered
+        using thread_mapper_callback_type =
+            util::function_nonser<bool(std::uint32_t)>;
 
         // thread-specific data
-        struct thread_data
+        class HPX_EXPORT thread_data
```

Comment:
I know this is in `detail` and the name doesn't matter too much, but would it be worth it to rename this to something like `os_thread_data` to avoid confusion with the HPX-thread `thread_data`?